optaplanner-core/src/main/java/org/optaplanner/core/api/solver/manager/SolverManager.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.api.solver.manager;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Consumer;
+
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.impl.solver.manager.DefaultSolverManager;
+
+public interface SolverManager<Solution_> extends AutoCloseable {
```

Comment:
This and the methods IMHO require Javadocs. Specifically, the fact that `problemId` is an `Object` begs for an explanation - what should I use for the id? What's the runtime behavior?

- What happens when I have two different instances of the id, but they `equals()`?)
- Is `null` accepted, where, and what is the behavior in that case?
- Can `null` be returned somewhere?

Reply:
The second point in the `Object problemId` is interesting and should have corresponding unit tests.